flink-kubernetes/src/test/java/org/apache/flink/kubernetes/utils/KubernetesUtilsTest.java

```diff
@@ -143,6 +146,21 @@ void testLoadPodFromTemplateAndCheckInitContainer() {
                 .isEqualTo(KubernetesPodTemplateTestUtils.createInitContainer());
     }
 
+    @Test
+    void testPodTemplateResourceLimitUtilisation() {
+        final FlinkPod flinkPod =
+                KubernetesUtils.loadPodFromTemplateFile(
+                        flinkKubeClient,
+                        KubernetesPodTemplateTestUtils.getPodTemplateFileWithResourceLimit(),
+                        KubernetesPodTemplateTestUtils.TESTING_MAIN_CONTAINER_NAME);
```

Comment:
We need to have tests for the different scenarios (covered in the logic):
 - limits specified but no memory in them
 - incorrect format in memory limit